README.md

```diff
@@ -1,29 +1,343 @@
+           // Error occurred. It's recommended you parse the `customerIOError` to learn more about the error.
+           break 
+         }
+       }
+     }
+     
+     func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+       MessagingPush.shared.application(application, didFailToRegisterForRemoteNotificationsWithError: error) { [weak self] result in 
+         // It's recommended to use `[weak self]` in the callback but your app's use cases may be unique. 
+         guard let self = self else { return }
+   
+         switch result {
+         case .success: 
+           // Device token successfully removed from profile!
+           break 
+         case .failure(let customerIOError):
+           // Error occurred. It's recommended you parse the `customerIOError` to learn more about the error.
+           break 
+         }
+       }
+     }
+   }
+   ```
+
+4. [Identify a customer](#Identify-a-customer) if you have not already. When you add a device token, it is not useful until you associate that device token with a person. You can identify a person before or after you register a device token with the Customer.io SDK. The SDK automatically adds and removes the device token from the customer profile when you identify and stop identifying a person with the SDK. 
+5. You should now be able to see a device token in your Customer.io Workspace for the identified person. You can send a simple push notification using the Customer.io push notification editor. If you want to use images, action buttons, or deep links you'll need to implement custom code in your app. 
+
+# Error handling 
+
+Whenever there's an error, the SDK returns a `CustomerIOError` instance. The Customer.io SDK at this time does not do error handling on behalf of your app - it's up to you. 
+
+ The `CustomerIOError` class helps you understand what went wrong and suggests how to handle the error.
+
+```swift
+let error: CustomerIOError = ...
+
+switch error {
+case .http(let httpError):
+    // An error happened while performing a HTTP request. 
+    // `httpError` is an instance of `HttpRequestError` and can also be parsed:
+    switch httpError {
+    ...
+    }
+    break
+case .notInitialized:
+    // SDK has not been initialized yet. Check the docs for `CustomerIO` class.
+    break 
+...
+}
+```
+
+To learn more about error handling, see all of the different types of cases of [the `CustomerIOError` class][s-cioerror]. Also, see our [example code giving suggestions on how to handle the various errors][s-cioerrorparse]. 
+
+# Testing 
+
+We designed the SDK with first-class support for automated testing, making it easy to inject dependencies and perform mocking in your code.
+
+## Dependency injection
+
+Every SDK class inherits from a Swift protocol. Inherited protocols use a consistent naming convention: `<NameOfClass>Instance`. For example, the `CustomerIO` class inherits the protocol `CustomerIOInstance`. 
```

Comment:
I didn't change this in my edit, but is "inherits from a Swift protocol" standard phrasing in the Swift world? I thought this should be "inherits a protocol from Swift", but didn't want to make a change in error.